yarn-project/ethereum/src/l1_tx_utils/readonly_l1_tx_utils.ts

```diff
@@ -144,14 +283,41 @@ export class ReadOnlyL1TxUtils {
       maxFeePerGas += priorityFee;
     }
 
-    // Ensure we don't exceed maxGwei
-    const maxGweiInWei = gasConfig.maxGwei! * WEI_CONST;
-    maxFeePerGas = maxFeePerGas > maxGweiInWei ? maxGweiInWei : maxFeePerGas;
+    // For replacement transactions, ensure we don't cap below the minimum required bump
+    // Otherwise the replacement will be rejected as underpriced
```

Comment:
I may be missing something, but isn't this handled above in L264 by this check, where `minPriorityFee` is the `minBump`?

```
      // Use maximum between competitive fee and minimum required bump
      const finalPriorityFee = competitivePriorityFee > minPriorityFee ? competitivePriorityFee : minPriorityFee;
```

Reply:
Comment isn't great, will update.
This is checking that our own imposed `maxGwei` doesn't impede the bump that was applied in the logic above.
I ran into this in e2e publisher tests where we cancel TXs.

I added a couple of things bc of this, the 'reasonable' price check & this but now I'm also thinking that `maxGwei` should not be set by default as it's mostly **not** used and the one time that it will be enforced, may end up costing the user without even knowing it was there